### PR TITLE
[xeclipse-150] fixed maven build script by using tycho

### PR DESCRIPTION
**Pull request by junhan**
Changes:
– add pom.xml to each top-level folder
– support generation of Linux + Windows for both x86 and x86 64 architectures
– add repository folder to keep the generated p2 repository
– add site folder to keep the generated update site
– add product folder to keep various products for different target platforms
– set all the version number of feature and products to 1.2.0.RC1

How to Test:
check out the repository
install maven 3.0.3
mvn clean install

### CODE REVIEW

File: features/org.xwiki.eclipse.feature/pom.xml [hunk 1]
```diff
@@ -1,96 +1,39 @@
...
+<project
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+  xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <parent>
-    <groupId>com.xpn.xwiki.platform</groupId>
-    <artifactId>xwiki</artifactId>
-    <version>14-SNAPSHOT</version>
+    <artifactId>org.xwiki.eclipse.features</artifactId>
+    <groupId>org.xwiki.eclipse</groupId>
+    <version>1.2.0.RC1</version>
```
Comment (vmassol) [inline features/org.xwiki.eclipse.feature/pom.xml]: Shouldn't this be 1.2.0-SNAPSHOT or 1.2-SNAPSHOT?

File: plugins/org.xwiki.eclipse.rcp/plugin.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@
          </property>
          <property
                name="aboutText"
-               value="XWiki Eclipse&#x0A;&#x0A;(C) 2007-2009 XWiki.com">
+               value="XWiki Eclipse&#x0A;&#x0A;(C) 2007-2008 XWiki.com">
```
Comment (vmassol) [inline plugins/org.xwiki.eclipse.rcp/plugin.xml]: This needs to be changed to "2007-2011 XWiki". Ideally the value should be computed automatically as we do in NOTICE.vm in XWiki Commons:

Copyright ${projectTimespan}#if($project.organization.name) $project.organization.name#else XWiki#end

File: features/org.xwiki.eclipse.feature/pom.xml (hunk 1 above)
Comment (junhan) [inline features/org.xwiki.eclipse.feature/pom.xml]: the version numbers have been changed to 1.2.0-SNAPSHOT in the commit of d24d37ae2a01e9a8366c50adcefeb1c3a2ef5868

File: plugins/org.xwiki.eclipse.rcp/plugin.xml (hunk 2 above)
Comment (junhan) [inline plugins/org.xwiki.eclipse.rcp/plugin.xml]: The about text are defined in two places (plugin.xml and product configurations). They are updated  in the commit of e0b9b407f1816df2c2f43fb7512d30e48724da4a

**Pull request merged**